Classes/EventPublisher/Mapping/DefaultMappingProvider.php

```diff
@@ -0,0 +1,222 @@
+<?php
+declare(strict_types=1);
+namespace Neos\EventSourcing\EventPublisher\Mapping;
+
+/*
+ * This file is part of the Neos.EventSourcing package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\EventSourcing\Event\DomainEventInterface;
+use Neos\EventSourcing\EventListener\EventListenerInterface;
+use Neos\EventSourcing\EventPublisher\Exception\EventPublisherException;
+use Neos\EventSourcing\EventPublisher\Exception\InvalidConfigurationException;
+use Neos\EventSourcing\EventPublisher\Exception\InvalidEventListenerException;
+use Neos\EventSourcing\EventStore\RawEvent;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Configuration\ConfigurationManager;
+use Neos\Flow\Configuration\Exception\InvalidConfigurationTypeException;
+use Neos\Flow\ObjectManagement\ObjectManagerInterface;
+use Neos\Flow\Reflection\ReflectionService;
+
+/**
+ * This factory for mappings will create Event Publisher mappings based on the configured EventStore.<store>.listeners
+ *
+ * @Flow\Scope("singleton")
+ */
+class DefaultMappingProvider
+{
+
+    /**
+     * @var Mappings[] indexed by the corresponding EventStore identifier
```

Comment:
Let's discuss the term "Mapping" one more time. I feel that something is still missing there, like "EventStoreToListenerMapping" or so...

Reply:
OK. I don't like to add the Event Store into the mix. Because it's basically just an "Event to Listener" Mapping (which is usually configured for an Event Store).
What do you think of `EventToListenerMapping`? And maybe it should belong to a different namespace? Maybe `Neos\EventSourcing\EventListener\EventToListenerMapping`?

Reply:
awesome yes :)